metomi/rose/config_processor.py

```diff
@@ -108,11 +106,11 @@ def __init__(self, event_handler=None, popen=None, fs_util=None):
         if fs_util is None:
             fs_util = FileSystemUtil(event_handler)
         self.fs_util = fs_util
-        path = os.path.dirname(
-            os.path.dirname(sys.modules["metomi.rose"].__file__)
-        )
         SchemeHandlersManager.__init__(
-            self, [path], "rose.config_processors", ["process"]
+            self,
+            [self.get_rose_path()],
+            "metomi.rose.config_processors",
+            ["process"]
         )
```

Comment:
I think `path` is now one level higher than it was before.

```python
>>> import metomi.rose
>>> import sys
>>> init = sys.modules["metomi.rose"].__file__
>>> init
'~/rose/metomi/rose/__init__.py'
>>> import os
>>> from pathlib import Path

# before
>>> os.path.dirname(os.path.dirname(init))
'~/rose/metomi'

# after
>>> Path(init).parent.parent.parent
PosixPath('~/rose')
```

If was `dirname(dirname(__init__.py)`, now it's `Path(__init__.py).parent.parent.parent`.



https://github.com/metomi/rose/blob/75ae8fdfd39d19d57898d5c24e7a485b92007fd8/metomi/rose/scheme_handler.py#L129-L139

Reply:
Yes, it was - I think that it's easier to understand if the next line has the full path.

Reply:
But doesn't this change the behaviour of SchemeHandlersManager?

Reply:
No

Reply:
I'm guessing it doesn't because of adding the `metomi.` to the beginning of the `ns` arg?